spec/models/async_transaction/base_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AsyncTransaction::Base, type: :model do
+  describe 'Validation' do
+    let(:transaction1) { create(:address_transaction) }
+    let(:transaction2) { build(:address_transaction) }
+    let(:transaction3) { build(:address_transaction, source: nil) }
+    let(:transaction4) { build(:address_transaction) }
+
+    it 'ensures uniqueness across source and transaction_id' do
+      transaction2.id = transaction1.id
+      expect(transaction2.valid?).to be false
+    end
+
+    it 'ensures presence of required fields' do
+      expect(transaction3.valid?).to be false
+    end
+
+    it 'accepts a FactoryBot-made transaction' do
+      expect(transaction4.valid?).to be true
+    end
+  end
+
+  describe 'Subclasses' do
+    let(:transaction1) { create(:address_transaction, transaction_id: 1) }
+    let(:transaction2) { create(:email_transaction, transaction_id: 2) }
+    let(:transaction3) { create(:telephone_transaction, transaction_id: 3) }
+
+    it 'are queryable from the parent' do
```

Comment:
For these `it` blocks that contain more than one expectation in them, you can add in `:aggregate_failures` to ensure that if one of the initial specs fail, the rest will still run.  Without it, if the first expectation fails, the rest are not run.

Here is an example:  https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/lib/evss/pciu/phone_number_spec.rb#L15